src/librustc/ty/query/plumbing.rs

```diff
@@ -153,8 +153,14 @@ impl<'a, 'tcx, Q: QueryDescription<'tcx>> JobOwner<'a, 'tcx, Q> {
             };
             mem::drop(lock);
 
-            if let Err(cycle) = job.await(tcx, span) {
-                return TryGetJob::JobCompleted(Err(cycle));
+            #[cfg(not(parallel_queries))]
+            return job.await(tcx, span);
+
+            #[cfg(parallel_queries)]
+            {
+                if let Err(cycle) = job.await(tcx, span) {
+                    return TryGetJob::JobCompleted(Err(cycle));
+                }
```

Comment:
I think the following would be more readable:
```rust
if cfg!(parallel_queries) {
    if let Err(cycle) = job.await(tcx, span) {
        return TryGetJob::JobCompleted(Err(cycle));
    }
} else {
    // If we get here in non-parallel mode we know that we have cycle error.
    let result = job.await(tcx, span);
    debug_assert!(result.is_err());
    return result;
}
```

Reply:
That doesn't work since `await` now has different signatures depending on `cfg!(parallel_queries)`. I'll add the comment though.

Reply:
:+1: